### PR TITLE
Fix zoom reset behavior when adding new connection

### PR DESCRIPTION
**Pull request by cheenamalhotra**
Fixes https://github.com/microsoft/azuredatastudio/issues/14128.

The problem is due to Angular/router always reloading the dashboard module components when navigating. I read about it and found some helpful articles which led me to try this solution, using `skipLocationChange` and `relativeTo` parameters.

Behavior with fix:

FixZoomReset

I have covered all router navigations, so this should fix the issue for all dashboard routes. The temporary workaround is also removed.

### CODE REVIEW

**Review by kburtram — APPROVED** [review]
Looks fine if we think that the Electron bump is resolving https://github.com/microsoft/azuredatastudio/issues/14128.  Otherwise, do we expect that bug to return with this update?  @alanrenmsft can you please take a look at this change?

File: src/sql/workbench/contrib/dashboard/browser/widgets/explorer/explorerWidget.component.ts [hunk 1]
```diff
@@ -46,6 +46,7 @@ export class ExplorerWidget extends DashboardWidget implements IDashboardWidget,
 
 	constructor(
 		@Inject(forwardRef(() => CommonServiceInterface)) private readonly _bootstrap: CommonServiceInterface,
+		@Inject(forwardRef(() => ActivatedRoute)) private activeRoute: ActivatedRoute,
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/dashboard/browser/widgets/explorer/explorerWidget.component.ts:49]: `private readonly _activeRoute`

File: src/sql/workbench/contrib/dashboard/browser/widgets/explorer/explorerTable.ts [hunk 2]
```diff
@@ -53,6 +53,7 @@ export class ExplorerTable extends Disposable {
 	private _propertiesToDisplay: ObjectListViewProperty[];
 
 	constructor(private parentElement: HTMLElement,
+		private activeRoute: ActivatedRoute,
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/dashboard/browser/widgets/explorer/explorerTable.ts:56]: `private readonly activeRoute`

File: src/sql/workbench/contrib/dashboard/browser/services/dashboardServiceInterface.service.ts [hunk 3]
```diff
@@ -69,6 +69,7 @@ export class DashboardServiceInterface extends CommonServiceInterface {
 		@Inject(IQueryManagementService) queryManagementService: IQueryManagementService,
 		@Inject(IBootstrapParams) params: IDashboardComponentParams,
 		@Inject(forwardRef(() => Router)) private _router: Router,
+		@Inject(forwardRef(() => ActivatedRoute)) private activeRoute: ActivatedRoute,
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/dashboard/browser/services/dashboardServiceInterface.service.ts:72]: `private _activeRoute`

File: src/sql/workbench/contrib/dashboard/browser/dashboard.component.ts [hunk 4]
```diff
@@ -31,6 +31,7 @@ export class DashboardComponent extends AngularDisposable implements OnInit {
 	constructor(
 		@Inject(forwardRef(() => CommonServiceInterface)) private _bootstrapService: CommonServiceInterface,
 		@Inject(forwardRef(() => Router)) private _router: Router,
+		@Inject(forwardRef(() => ActivatedRoute)) private activeRoute: ActivatedRoute,
```
Comment (Charles-Gagnon) [inline src/sql/workbench/contrib/dashboard/browser/dashboard.component.ts:34]: `private _activeRoute`

File: src/sql/base/browser/ui/breadcrumb/breadcrumb.component.ts [hunk 5]
```diff
@@ -36,6 +36,7 @@ export class BreadcrumbComponent implements OnInit, OnDestroy {
 	constructor(
 		@Inject(forwardRef(() => IBreadcrumbService)) private _breadcrumbService: IBreadcrumbService,
 		@Inject(forwardRef(() => Router)) private _router: Router,
+		@Inject(forwardRef(() => ActivatedRoute)) private activeRoute: ActivatedRoute,
```
Comment (Charles-Gagnon) [inline src/sql/base/browser/ui/breadcrumb/breadcrumb.component.ts:39]: Suggest keeping naming/modifier conventions in the files

`private _activeRoute`

(downside of the current form of suggestions, renaming stuff is annoying if it's used in multiple places....)

**Review by Charles-Gagnon — APPROVED** [review]
❤️ Awesome!

Could you link to the articles you used to find this out?

**Comment by cheenamalhotra** [comment]
> ❤️ Awesome!
> 
> Could you link to the articles you used to find this out?

The first thing I understood after debugging the behavior, was that this was occurring from Angular/router package during module refresh/reload - that led to zoom reset, so that had to be prevented. I found [this post](https://stackoverflow.com/questions/55577179/angular7-router-is-refreshing-entire-screen) that hinted me that it might be happening when calling router.navigate API.

So, I experimented with router.navigate API options and wanted to use relativeTo option but that didn't work alone, so I found this article that made things clear: [Angular Router: Revealing some interesting facts and features](https://indepth.dev/posts/1379/angular-router-revealing-some-interesting-facts-and-features). That led me to **skipLocationChange** to be used alongside and testing that made the difference 🙂

**Comment by alanrenmsft** [comment]
@cheenamalhotra thanks for looking into the issue. I read the links you posted, but it doesn't mention Electron at all, how does the Angular interfere with Electron?

**Comment by alanrenmsft** [comment]
have you tried just removing the workaround that I added? maybe it is something that got fixed with the new Electron version?

**Comment by cheenamalhotra** [comment]
> have you tried just removing the workaround that I added? maybe it is something that got fixed with the new Electron version?

I did try that, but that did not fix but brought back the original behavior of reset. 
I had to provide the parameters to router navigate to get it to working.

**Comment by alanrenmsft** [comment]
ok, it is good that the issue is fixed, but we don't know why the zoom level of Electron was reset because of something happened in Angular.

**Review by alanrenmsft — APPROVED** [review]

**Pull request merged**